%% LyX 2.3.7 created this file.  For more info, see http://www.lyx.org/.
%% Do not edit unless you really know what you are doing.
\documentclass[journal,article,submit,pdftex,moreauthors]{Definitions/mdpi}
\usepackage[utf8]{inputenc}
\usepackage{array}
\usepackage{float}
\usepackage{booktabs}
\usepackage{textcomp}
\usepackage{url}
\ifx\hypersetup\undefined
  \AtBeginDocument{%
    \hypersetup{unicode=true}
  }
\else
  \hypersetup{unicode=true}
\fi

\makeatletter

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%% LyX specific LaTeX commands.

\Title{EEGO: an extended version of Eel and grouper optimizer for global
optimization problems}

\TitleCitation{EEGO: an extended version of Eel and grouper optimizer for global
optimization problems}

\Author{Glykeria Kyrou$^{1}$, Vasileios Charilogis$^{2}$ and Ioannis G.
Tsoulos$^{3,*}$}

\AuthorNames{Glykeria Kyrou, Vasileios Charilogis and Ioannis G. Tsoulos }

\AuthorCitation{Kyrou, G.; Charilogis, V.; Tsoulos, I.G. }


\address{$^{1}$\quad{}Department of Informatics and Telecommunications,
University of Ioannina, 47150 Kostaki Artas, Greece; g.kyrou@uoi.gr\\
$^{2}$\quad{}Department of Informatics and Telecommunications, University
of Ioannina, 47150 Kostaki Artas, Greece; v.charilog@uoi.gr\\
$^{3}\quad$Department of Informatics and Telecommunications, University
of Ioannina, 47150 Kostaki Artas, Greece;itsoulos@uoi.gr}


\corres{Correspondence: itsoulos@uoi.gr}


\abstract{The problems of finding a global minimum of a function are increasingly
applied to real-world problems. As a result, a variety of computational
techniques have been developed to better locate the global minimum.
A decisive role is played by evolutionary techniques, which simulate
natural processes and aim to find the global minimum of multidimensional
functions. A recently introduced evolutionary technique is the optimal
Eel and Grouper (EGO) algorithm, which is inspired by the symbiotic
interaction and foraging strategy of eels and groupers in marine ecosystems.
The EGO algorithm is characterized for its reliability in locating
the global minimum. In this paper, modifications are proposed that
aim to improve the reliability and speed of the above technique, such
as the application of a termination technique based on stochastic
observations and an innovative sampling method. The proposed method
was tested on several problems from the relevant literature and a
comparative study was made with other global optimization techniques
with promising results.}


\keyword{Global optimization; Meta-heuristic; Stochastic techniques; Swarm
based methods}

%% Because html converters don't know tabularnewline
\providecommand{\tabularnewline}{\\}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%% User specified LaTeX commands.
%  LaTeX support: latex@mdpi.com 
%  For support, please attach all files needed for compiling as well as the log file, and specify your operating system, LaTeX version, and LaTeX editor.

%=================================================================


% For posting an early version of this manuscript as a preprint, you may use "preprints" as the journal and change "submit" to "accept". The document class line would be, e.g., \documentclass[preprints,article,accept,moreauthors,pdftex]{mdpi}. This is especially recommended for submission to arXiv, where line numbers should be removed before posting. For preprints.org, the editorial staff will make this change immediately prior to posting.

%--------------------
% Class Options:
%--------------------
%----------
% journal
%----------
% Choose between the following MDPI journals:
% acoustics, actuators, addictions, admsci, adolescents, aerospace, agriculture, agriengineering, agronomy, ai, algorithms, allergies, alloys, analytica, animals, antibiotics, antibodies, antioxidants, applbiosci, appliedchem, appliedmath, applmech, applmicrobiol, applnano, applsci, aquacj, architecture, arts, asc, asi, astronomy, atmosphere, atoms, audiolres, automation, axioms, bacteria, batteries, bdcc, behavsci, beverages, biochem, bioengineering, biologics, biology, biomass, biomechanics, biomed, biomedicines, biomedinformatics, biomimetics, biomolecules, biophysica, biosensors, biotech, birds, bloods, blsf, brainsci, breath, buildings, businesses, cancers, carbon, cardiogenetics, catalysts, cells, ceramics, challenges, chemengineering, chemistry, chemosensors, chemproc, children, chips, cimb, civileng, cleantechnol, climate, clinpract, clockssleep, cmd, coasts, coatings, colloids, colorants, commodities, compounds, computation, computers, condensedmatter, conservation, constrmater, cosmetics, covid, crops, cryptography, crystals, csmf, ctn, curroncol, currophthalmol, cyber, dairy, data, dentistry, dermato, dermatopathology, designs, diabetology, diagnostics, dietetics, digital, disabilities, diseases, diversity, dna, drones, dynamics, earth, ebj, ecologies, econometrics, economies, education, ejihpe, electricity, electrochem, electronicmat, electronics, encyclopedia, endocrines, energies, eng, engproc, ent, entomology, entropy, environments, environsciproc, epidemiologia, epigenomes, est, fermentation, fibers, fintech, fire, fishes, fluids, foods, forecasting, forensicsci, forests, foundations, fractalfract, fuels, futureinternet, futureparasites, futurepharmacol, futurephys, futuretransp, galaxies, games, gases, gastroent, gastrointestdisord, gels, genealogy, genes, geographies, geohazards, geomatics, geosciences, geotechnics, geriatrics, hazardousmatters, healthcare, hearts, hemato, heritage, highthroughput, histories, horticulturae, humanities, humans, hydrobiology, hydrogen, hydrology, hygiene, idr, ijerph, ijfs, ijgi, ijms, ijns, ijtm, ijtpp, immuno, informatics, information, infrastructures, inorganics, insects, instruments, inventions, iot, j, jal, jcdd, jcm, jcp, jcs, jdb, jeta, jfb, jfmk, jimaging, jintelligence, jlpea, jmmp, jmp, jmse, jne, jnt, jof, joitmc, jor, journalmedia, jox, jpm, jrfm, jsan, jtaer, jzbg, kidney, kidneydial, knowledge, land, languages, laws, life, liquids, literature, livers, logics, logistics, lubricants, lymphatics, machines, macromol, magnetism, magnetochemistry, make, marinedrugs, materials, materproc, mathematics, mca, measurements, medicina, medicines, medsci, membranes, merits, metabolites, metals, meteorology, methane, metrology, micro, microarrays, microbiolres, micromachines, microorganisms, microplastics, minerals, mining, modelling, molbank, molecules, mps, msf, mti, muscles, nanoenergyadv, nanomanufacturing, nanomaterials, ncrna, network, neuroglia, neurolint, neurosci, nitrogen, notspecified, nri, nursrep, nutraceuticals, nutrients, obesities, oceans, ohbm, onco, oncopathology, optics, oral, organics, organoids, osteology, oxygen, parasites, parasitologia, particles, pathogens, pathophysiology, pediatrrep, pharmaceuticals, pharmaceutics, pharmacoepidemiology, pharmacy, philosophies, photochem, photonics, phycology, physchem, physics, physiologia, plants, plasma, pollutants, polymers, polysaccharides, poultry, powders, preprints, proceedings, processes, prosthesis, proteomes, psf, psych, psychiatryint, psychoactives, publications, quantumrep, quaternary, qubs, radiation, reactions, recycling, regeneration, religions, remotesensing, reports, reprodmed, resources, rheumato, risks, robotics, ruminants, safety, sci, scipharm, seeds, sensors, separations, sexes, signals, sinusitis, skins, smartcities, sna, societies, socsci, software, soilsystems, solar, solids, sports, standards, stats, stresses, surfaces, surgeries, suschem, sustainability, symmetry, synbio, systems, taxonomy, technologies, telecom, test, textiles, thalassrep, thermo, tomography, tourismhosp, toxics, toxins, transplantology, transportation, traumacare, traumas, tropicalmed, universe, urbansci, uro, vaccines, vehicles, venereology, vetsci, vibration, viruses, vision, waste, water, wem, wevj, wind, women, world, youth, zoonoticdis 

%---------
% article
%---------
% The default type of manuscript is "article", but can be replaced by: 
% abstract, addendum, article, book, bookreview, briefreport, casereport, comment, commentary, communication, conferenceproceedings, correction, conferencereport, entry, expressionofconcern, extendedabstract, datadescriptor, editorial, essay, erratum, hypothesis, interestingimage, obituary, opinion, projectreport, reply, retraction, review, perspective, protocol, shortnote, studyprotocol, systematicreview, supfile, technicalnote, viewpoint, guidelines, registeredreport, tutorial
% supfile = supplementary materials

%----------
% submit
%----------
% The class option "submit" will be changed to "accept" by the Editorial Office when the paper is accepted. This will only make changes to the frontpage (e.g., the logo of the journal will get visible), the headings, and the copyright information. Also, line numbering will be removed. Journal info and pagination for accepted papers will also be assigned by the Editorial Office.

%------------------
% moreauthors
%------------------
% If there is only one author the class option oneauthor should be used. Otherwise use the class option moreauthors.

%---------
% pdftex
%---------
% The option pdftex is for use with pdfLaTeX. If eps figures are used, remove the option pdftex and use LaTeX and dvi2pdf.

%=================================================================
% MDPI internal commands - do not modify
\firstpage{1} 
 
\setcounter{page}{\@firstpage} 

\pubvolume{1}
\issuenum{1}
\articlenumber{0}
\pubyear{2024}
\copyrightyear{2024}
%\externaleditor{Academic Editor: Firstname Lastname} % For journal Automation, please change Academic Editor to "Communicated by"
\datereceived{}
\daterevised{ } % Comment out if no revised date
\dateaccepted{}
\datepublished{}
%\datecorrected{} % Corrected papers include a "Corrected: XXX" date in the original paper.
%\dateretracted{} % Corrected papers include a "Retracted: XXX" date in the original paper.
\hreflink{https://doi.org/} % If needed use \linebreak
%\doinum{}
%------------------------------------------------------------------
% The following line should be uncommented if the LaTeX file is uploaded to arXiv.org
%\pdfoutput=1

%=================================================================
% Add packages and commands here. The following packages are loaded in our class file: fontenc, inputenc, calc, indentfirst, fancyhdr, graphicx, epstopdf, lastpage, ifthen, lineno, float, amsmath, setspace, enumitem, mathpazo, booktabs, titlesec, etoolbox, tabto, xcolor, soul, multirow, microtype, tikz, totcount, changepage, attrib, upgreek, cleveref, amsthm, hyphenat, natbib, hyperref, footmisc, url, geometry, newfloat, caption

%=================================================================
%% Please use the following mathematics environments: Theorem, Lemma, Corollary, Proposition, Characterization, Property, Problem, Example, ExamplesandDefinitions, Hypothesis, Remark, Definition, Notation, Assumption
%% For proofs, please use the proof environment (the amsthm package is loaded by the MDPI class).

%=================================================================
% The fields PACS, MSC, and JEL may be left empty or commented out if not applicable
%\PACS{J0101}
%\MSC{}
%\JEL{}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% Only for the journal Diversity
%\LSID{\url{http://}}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% Only for the journal Applied Sciences:
%\featuredapplication{Authors are encouraged to provide a concise description of the specific application or a potential application of the work. This section is not mandatory.}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% Only for the journal Data:
%\dataset{DOI number or link to the deposited data set in cases where the data set is published or set to be published separately. If the data set is submitted and will be published as a supplement to this paper in the journal Data, this field will be filled by the editors of the journal. In this case, please make sure to submit the data set as a supplement when entering your manuscript into our manuscript editorial system.}

%\datasetlicense{license under which the data set is made available (CC0, CC-BY, CC-BY-SA, CC-BY-NC, etc.)}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% Only for the journal Toxins
%\keycontribution{The breakthroughs or highlights of the manuscript. Authors can write one or two sentences to describe the most important part of the paper.}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% Only for the journal Encyclopedia
%\encyclopediadef{Instead of the abstract}
%\entrylink{The Link to this entry published on the encyclopedia platform.}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% Only for the journal Advances in Respiratory Medicine
%\addhighlights{yes}
%\renewcommand{\addhighlights}{%

%\noindent This is an obligatory section in â€œAdvances in Respiratory Medicineâ€, whose goal is to increase the discoverability and readability of the article via search engines and other scholars. Highlights should not be a copy of the abstract, but a simple text allowing the reader to quickly and simplified find out what the article is about and what can be cited from it. Each of these parts should be devoted up to 2~bullet points.\vspace{3pt}\\
%\textbf{What are the main findings?}
% \begin{itemize}[labelsep=2.5mm,topsep=-3pt]
% \item First bullet.
% \item Second bullet.
% \end{itemize}\vspace{3pt}
%\textbf{What is the implication of the main finding?}
% \begin{itemize}[labelsep=2.5mm,topsep=-3pt]
% \item First bullet.
% \item Second bullet.
% \end{itemize}
%}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\makeatother

\begin{document}
\maketitle

\section{Introduction}

The basic goal of global optimization is to find the global minimum
by searching for the appropriate scope of each problem. Primarily,
a global optimization method aims to discover the global minimum of
a continuous function, and it is defined as
\begin{equation}
x^{*}=\mbox{arg}\min_{x\in S}f(x)\label{eq:eq1}
\end{equation}
with $S$: 
\[
S=\left[a_{1},b_{1}\right]\times\left[a_{2},b_{2}\right]\times\ldots\left[a_{n},b_{n}\right]
\]

Global optimization is an integral part of many areas of our daily
lives. One of these fields is computer science\citep{plhroforikh,computer,computer1}.
In addition to the computer field, global optimization finds application
in sciences such as mathematics\citep{maths,maths-1,maths2,key-maths3},
physics\citep{fusikh,fusikh1,fysikhh}, chemistry\citep{xhmeia,xhmeia1,xhmeia2}
and medicine\citep{iatrikh,iatrikh1,medicine}. Optimization methods
are divided into two categories, deterministic\citep{determistic,determistic1,determistic2}
and stochastic\citep{stohastic,stohastic1,stohastic2}. The first
category, i.e. that of deterministic methods, has as its main objective
the identification of the overall optimal solution and is mainly used
in simple problems. On the contrary, stochastic methods are mainly
used in complex problems.

Swarm Intelligence Algorithms\citep{swarm1,swarm2,swarm,swarm3,swarm4}
as a source of inspiration and collective behavior of insects and
other animals. Intelligence Algorithms mimic systems in which agents
interact locally and cooperate worldwide to solve optimization problems.
Swarm intelligence algorithms are very important tools for dealing
with complex optimization problems in a wide range of applications\citep{APPS}.
Characteristic examples of such algorithms are the WOA algorithm \citealp{WOA,WOA1,WOA2,ewoa,woa-ssa,woaa},
SCA algorithm \citep{SCA,SCA1,SCA2,ssaa,sca1}and SSA algorithm\citep{SSA,SSA1,SSA2,ssaa,woa-ssa}.

The EGO algorithm is a reliable technique for real-world optimization
problems. The EGO algorithm is inspired by the symbiotic interaction
and foraging strategy of eels and groupers in marine ecosystems. In
nature, there are complex interactions between biological species.
Relationships of this type are classified into five categories\citep{search algorithm,several metaheuristic algorithms }:1.
Naturalism: Where two species can live in an ecosystem without affecting
each other. 2. In predation, where one creature dies to feed another.
3. Parasitism: where one species causes harm to another without always
killing it. 4. In competitive mode, the same or different organizations
compete for resources. 5. Mutualism: \citep{mutualism-parasitism,mutualistic,Competition in mutualistic systems}when
two organisms have a beneficial INTERACTION. Bshary et al. \citep{hunting between groupers and giant moray eels in the Red Sea}consider
that target ingestion, something observed in eels and groupers, is
a necessary condition for interspecific cooperative hunting to occur.
Intraspecific predation could increase the hunting efficiency of predators
by mammals. According to Ali Mohammadzadeh and Seyedali Mirjalili
the EGO optimization algorithm \citep{Eel and grouper optimizer}generates
a set of random answers, then stores the best answers found so far,
allocates them to the target point, and changes the answers with them.
As the number of iterations increases, the limits of the sine function
are changed to enhance the phase of finding the best solution. This
method stops the process when the iteration exceeds the maximum number.
Because the EGO optimization algorithm generates and boosts a collection
of random responses, it has the advantage of increased local optimum
discovery and avoidance compared to individual methods.

This paper introduces some modifications to the EGO algorithm in order
to improve its efficiency. The proposed amendments are presented below:
\begin{itemize}
\item The addition of a sampling technique based on the K-means method\citep{kmeans-1,kmeans2,kmeans3,kmeans4,kmeans-ereunhtikh-koinothta,kmeans-paterrn}.
The sampled points will help to find the global minimum of the function
in the most efficient way. Additionally, by using this method, points
that are in proximity can be rejected.
\item The use of a termination technique based on stochastic observations.
At each iteration of the algorithm, the smallest value is recorded.
Once it remains stable for a predetermined number of iterations, the
method terminates. The present termination method helps with the fastest
termination without unnecessarily wasting computing time.
\item Using mod1 and mod2 and mod3 parameters which take values of either
1, which means it is Disable or 2, which means it is Enable. mod1
specifies how randomness will be used when updating solutions. When
mod1 is set to 1 p, it takes values from 0 to 1, while when mod1 is
set to 2 p it takes values from -1.0 to 1.0. The mod2 parameter affects
how the variables f1 and f2 are defined, which in turn affects the
calculation of the new positions. The mod3 parameter determines how
the algorithm handles thresholds when agent positions exceed predefined
minimum and maximum thresholds. When mod3 is set to 1, it ensures
that all positions remain within valid limits. When mod3 is set to
2 these positions are not taken into account and the fish wait for
better positions to attack.
\end{itemize}
The rest of this paper is divided into the following sections: in
section \ref{sec:Materials-and-Methods}, the proposed method is fully
described.

\section{The proposed method\label{sec:Materials-and-Methods}}

\subsection{The main steps of the algorithm}

The main steps of the used global optimization method are the following:
\begin{enumerate}
\item \textbf{Initialization step}.
\begin{enumerate}
\item Define as $N_{c}$ the number of elements in the search Agents
\item Define as $N_{g}$, the maximum number of allowed iterations.
\item Initialize randomly the members of the search Agents in set $S$. 
\end{enumerate}
\item \textbf{Calculation step}.
\end{enumerate}
\begin{verse}
(a) While $\left[\left(t\prec Ng\right)\right]$

(b) Update Variables a and starrvation\_rate where starrvation\_rate
is a number in {[}0,100{]}. According to types:
\begin{itemize}
\item $a=2-2*(t\,/\,Ng)$ 
\item $\left(starvatiorate=100*(t/\,Ng)\right)$
\end{itemize}
(c) Compute the fitness of each search Agents.

(d) Sort all solutions in the current population fro the best to worst
according to the function value.
\begin{quote}
\textbf{For} $j=1,\ldots,N_{c}$ \textbf{do}
\end{quote}
\begin{itemize}
\item Update variables r1, r2, r3, r4, C1, C2 and p : where $r_{1},r_{2}$
are random numbers in $,[0,1]$ , C1 is a rondom number in {[}-a,a{]},
C2 is a random number in {[}0,2{]}. According to types:
\begin{itemize}
\item $r3=(a-2)*r1+2$ 
\item $r4=100*r2$ 
\item $C1=2*a*r1-a$
\item $C2=2*r1$ 
\item $b=a*r2$
\end{itemize}
\item Update the position of each search Agent:
\item $if(r4<=starvation_{r}ate)$
\item if mod1 has the value 1
\begin{itemize}
\item p is a random number in$\;[0.0,1.0]$
\end{itemize}
\item if mod1 has the value 2 
\begin{itemize}
\item p is a random number in$\;[-1.0,1.0]$
\end{itemize}
\item if mod2 has the value 1
\begin{itemize}
\item $f1=0.8$
\item $f2=0.2$
\end{itemize}
\item if mod2 has the value 2 
\begin{itemize}
\item f1 is a random number in$\;[0.0,2.0]$ 
\item f1 is a random number in$\;[-2.0,0.0]$
\end{itemize}
\end{itemize}
\begin{quote}
\textbf{If $\left(p\prec0.5\right)$}
\begin{itemize}
\item Change the location of current search by :$\left[\left(f1\ast X1\dotplus f2\ast X2\right)\not\:\:2\right]$ 
\end{itemize}
else if$\:\left(p\geq0.5\right)$
\begin{itemize}
\item Change the location of current search by :$\left(f2\ast X1\dotplus f1\ast X2\not\:\:2\right)$
\end{itemize}
\textbf{End if}

\textbf{End For}
\end{quote}
(e) Ensure No Agents leave the search area

(f) Evaluate the performance of each search Agents

(g) Update XPrey 
\begin{itemize}
\item if there is a better solution
\end{itemize}
(h)\textbf{ }Set $t=t+1$

(i) End while

(j) Return the best solution XPrey
\end{verse}
\begin{description}
\item [{3.}] \textbf{Termination check step}

\textbf{(a) Set} $t=t+1$

\textbf{(b) Calculate} to stopping rule in the work of Charilogis
\citep{charilogis}

\textbf{(c) If} the termination criteria are not met then go to Calculation
step, \textbf{else} terminate.
\end{description}

\subsection{The proposed sampling procedure\label{subsec:The-proposed-sampling}}

The sampling technique applied in this work initially generates samples
from the objective function. Then, through the K-means method, only
the recognized centers are selected as final samples. This technique,
which is an achievement of James MacQueen {[}41{]}, is one of the
most well-known clustering algorithms in the broad research community,
both in data analysis and in machine learning {[}42, 43{]} and pattern
recognition\citep{kmeans-paterrn}. The algorithm mainly aims to estimate
the centers of possible groups from a set of samples. Next, the basic
steps of the algorithm are presented:
\begin{enumerate}
\item \textbf{Define} as $k$ the number of clusters.
\item Randomly select $N_{m}$ initial points $x_{i},\ i=1,\ldots,N_{m}$
from the objective function.
\item Randomly assign each point $x_{i},\ i=1,...,N_{m}$ in a cluster $S_{j},\ j=1,\ldots,k$.
\item \textbf{For} every cluster $j=1..k$ \textbf{do}
\begin{itemize}
\item \textbf{Set} as $M_{j}$ the number of points in $S_{j}$
\item \textbf{Calculate }the center of the cluster $c_{j}$ as
\[
c_{j}=\frac{1}{M_{j}}\sum_{x_{i}\in S_{j}}x_{i}
\]
\end{itemize}
\item \textbf{End For}
\item \textbf{Repeat the following steps:}
\begin{itemize}
\item Set $S_{j}=\left\{ \right\} ,\ j=1..k$
\item \textbf{For} each point $x_{i},\ i=1,...,N_{m}$ \textbf{do}
\begin{itemize}
\item \textbf{Set} $j^{*}=\mbox{argmin}_{m=1}^{k}\left\{ D\left(x_{i},c_{m}\right)\right\} $.
The function $D(x,y)$ is the Euclidean distance of points $(x,y)$.
\item \textbf{Set} $S_{j^{*}}=S_{j^{*}}\cup\left\{ x_{i}\right\} $.
\end{itemize}
\item \textbf{End For}
\item \textbf{For} each center $c_{j},\ j=1..k$ \textbf{do}
\begin{itemize}
\item \textbf{Update }the center $c_{j}$ as
\[
c_{j}=\frac{1}{M_{j}}\sum_{x_{i}\in S_{j}}x_{i}
\]
\end{itemize}
\item \textbf{End For}
\end{itemize}
\item \textbf{If }there is no significant change in centers $c_{j}$\textbf{
terminate }the algorithm and return the $k$ centers as the final
set of samples.
\end{enumerate}

\section{Results\label{sec:Results}}

This section will begin with a detailed description of the functions
that will be used in the experiments, followed by an analysis of the
experiments performed and comparisons with other global optimization
techniques. 

\subsection{Test functions }

The functions used in the experiments have been proposed in a series
of relative works\citep{Ali,Floudas1} and they cover various scientific
fields, such as medicine, physics, engineering, etc. Also, these objective
functions have been used by many researchers in a variety of publications.\citep{testfunc1,testfunc2,testfunc2-1,testfunc3,testfunc4}
The definitions of these functions are given below:
\begin{itemize}
\item \textbf{Bf1} (Bohachevsky 1) function:
\end{itemize}
\[
f(x)=x_{1}^{2}+2x_{2}^{2}-\frac{3}{10}\cos\left(3\pi x_{1}\right)-\frac{4}{10}\cos\left(4\pi x_{2}\right)+\frac{7}{10}
\]

\begin{itemize}
\item \textbf{Bf2} (Bohachevsky 2) function: 
\[
f(x)=x_{1}^{2}+2x_{2}^{2}-\frac{3}{10}\cos\left(3\pi x_{1}\right)\cos\left(4\pi x_{2}\right)+\frac{3}{10}
\]
\item \textbf{Bf3} (Bohachevsky 3) function: 
\[
f(x)=x_{1}^{2}+2x_{2}^{2}-\frac{3}{10}\cos\left(3\pi x_{1}+4\pi x_{2}\right)+\frac{3}{10}
\]
\item \textbf{Branin} function:

\[
f(x)=\left(x_{2}-\frac{5.1}{4\pi^{2}}x_{1}^{2}+\frac{5}{\pi}x_{1}-6\right)^{2}+10\left(1-\frac{1}{8\pi}\right)\cos(x_{1})+10
\]
 with $-5\le x_{1}\le10,\ 0\le x_{2}\le15$.
\item \textbf{Camel} function:
\[
f(x)=4x_{1}^{2}-2.1x_{1}^{4}+\frac{1}{3}x_{1}^{6}+x_{1}x_{2}-4x_{2}^{2}+4x_{2}^{4},\quad x\in[-5,5]^{2}
\]
\item \textbf{Easom} function: 
\[
f(x)=-\cos\left(x_{1}\right)\cos\left(x_{2}\right)\exp\left(\left(x_{2}-\pi\right)^{2}-\left(x_{1}-\pi\right)^{2}\right)
\]
with $x\in[-100,100]^{2}.$ 
\item \textbf{Exponential} function, defined as: 
\[
f(x)=-\exp\left(-0.5\sum_{i=1}^{n}x_{i}^{2}\right),\quad-1\le x_{i}\le1
\]
 In the conducted experiments the values $n=4,8,16,32$ were used.
\item \textbf{Griewank2} function:
\[
f(x)=1+\frac{1}{200}\sum_{i=1}^{2}x_{i}^{2}-\prod_{i=1}^{2}\frac{\cos(x_{i})}{\sqrt{(i)}},\quad x\in[-100,100]^{2}
\]
\item \textbf{Griewank10} function. The function is given by the equation
\[
f(x)=\sum_{i=1}^{n}\frac{x_{i}^{2}}{4000}-\prod_{i=1}^{n}\cos\left(\frac{x_{i}}{\sqrt{i}}\right)+1
\]
with $n=10$.
\item \textbf{Gkls} function\citep{gkls}. $f(x)=\mbox{Gkls}(x,n,w)$, is
a constructed function with $w$ local minima presented in \citep{gkls},
with $x\in[-1,1]^{n}$. For the conducted experiments the values $n=2,3$
and $w=50$ were utilized.
\item \textbf{Goldstein and Price function }\\
\begin{eqnarray*}
f(x) & = & \left[1+\left(x_{1}+x_{2}+1\right)^{2}\right.\\
 &  & \left(19-14x_{1}+3x_{1}^{2}-14x_{2}+6x_{1}x_{2}+3x_{2}^{2}\right)]\times\\
 &  & [30+\left(2x_{1}-3x_{2}\right)^{2}\\
 &  & \left(18-32x_{1}+12x_{1}^{2}+48x_{2}-36x_{1}x_{2}+27x_{2}^{2}\right)]
\end{eqnarray*}
With $x\in[-2,2]^{2}$. 
\item \textbf{Hansen} function: $f(x)=\sum_{i=1}^{5}i\cos\left[(i-1)x_{1}+i\right]\sum_{j=1}^{5}j\cos\left[(j+1)x_{2}+j\right]$,
$x\in[-10,10]^{2}$ .
\item \textbf{Hartman 3} function:
\[
f(x)=-\sum_{i=1}^{4}c_{i}\exp\left(-\sum_{j=1}^{3}a_{ij}\left(x_{j}-p_{ij}\right)^{2}\right)
\]
with $x\in[0,1]^{3}$ and $a=\left(\begin{array}{ccc}
3 & 10 & 30\\
0.1 & 10 & 35\\
3 & 10 & 30\\
0.1 & 10 & 35
\end{array}\right),\ c=\left(\begin{array}{c}
1\\
1.2\\
3\\
3.2
\end{array}\right)$ and
\[
p=\left(\begin{array}{ccc}
0.3689 & 0.117 & 0.2673\\
0.4699 & 0.4387 & 0.747\\
0.1091 & 0.8732 & 0.5547\\
0.03815 & 0.5743 & 0.8828
\end{array}\right)
\]
\item \textbf{Hartman 6} function:
\[
f(x)=-\sum_{i=1}^{4}c_{i}\exp\left(-\sum_{j=1}^{6}a_{ij}\left(x_{j}-p_{ij}\right)^{2}\right)
\]
with $x\in[0,1]^{6}$ and $a=\left(\begin{array}{cccccc}
10 & 3 & 17 & 3.5 & 1.7 & 8\\
0.05 & 10 & 17 & 0.1 & 8 & 14\\
3 & 3.5 & 1.7 & 10 & 17 & 8\\
17 & 8 & 0.05 & 10 & 0.1 & 14
\end{array}\right),\ c=\left(\begin{array}{c}
1\\
1.2\\
3\\
3.2
\end{array}\right)$ and
\[
p=\left(\begin{array}{cccccc}
0.1312 & 0.1696 & 0.5569 & 0.0124 & 0.8283 & 0.5886\\
0.2329 & 0.4135 & 0.8307 & 0.3736 & 0.1004 & 0.9991\\
0.2348 & 0.1451 & 0.3522 & 0.2883 & 0.3047 & 0.6650\\
0.4047 & 0.8828 & 0.8732 & 0.5743 & 0.1091 & 0.0381
\end{array}\right)
\]
\item \textbf{Potential} function, this function stands for the energy of
a molecular conformation of N atoms, that interacts using via the
Lennard-Jones potential\citep{jones}. The function is defined as:
\[
V_{LJ}(r)=4\epsilon\left[\left(\frac{\sigma}{r}\right)^{12}-\left(\frac{\sigma}{r}\right)^{6}\right]
\]
For the conducted experiments the values $N=3,\ 5$ were used. 
\item \textbf{Rastrigin} function. 
\[
f(x)=x_{1}^{2}+x_{2}^{2}-\cos(18x_{1})-\cos(18x_{2}),\quad x\in[-1,1]^{2}
\]
\item \textbf{\emph{Rosenbrock}}\emph{ function}.\\
\[
f(x)=\sum_{i=1}^{n-1}\left(100\left(x_{i+1}-x_{i}^{2}\right)^{2}+\left(x_{i}-1\right)^{2}\right),\quad-30\le x_{i}\le30.
\]
The values $n=4,\ 8,\ 16$ were used in the conducted experiments.
\item \textbf{Shekel 5 }function.
\end{itemize}
\[
f(x)=-\sum_{i=1}^{5}\frac{1}{(x-a_{i})(x-a_{i})^{T}+c_{i}}
\]
 

with $x\in[0,10]^{4}$ and $a=\left(\begin{array}{cccc}
4 & 4 & 4 & 4\\
1 & 1 & 1 & 1\\
8 & 8 & 8 & 8\\
6 & 6 & 6 & 6\\
3 & 7 & 3 & 7
\end{array}\right),\ c=\left(\begin{array}{c}
0.1\\
0.2\\
0.2\\
0.4\\
0.4
\end{array}\right)$
\begin{itemize}
\item \textbf{Shekel 7} function.
\end{itemize}
\[
f(x)=-\sum_{i=1}^{7}\frac{1}{(x-a_{i})(x-a_{i})^{T}+c_{i}}
\]

with $x\in[0,10]^{4}$ and $a=\left(\begin{array}{cccc}
4 & 4 & 4 & 4\\
1 & 1 & 1 & 1\\
8 & 8 & 8 & 8\\
6 & 6 & 6 & 6\\
3 & 7 & 3 & 7\\
2 & 9 & 2 & 9\\
5 & 3 & 5 & 3
\end{array}\right),\ c=\left(\begin{array}{c}
0.1\\
0.2\\
0.2\\
0.4\\
0.4\\
0.6\\
0.3
\end{array}\right)$.
\begin{itemize}
\item \textbf{Shekel 10} function.
\end{itemize}
\[
f(x)=-\sum_{i=1}^{10}\frac{1}{(x-a_{i})(x-a_{i})^{T}+c_{i}}
\]
 

with $x\in[0,10]^{4}$ and $a=\left(\begin{array}{cccc}
4 & 4 & 4 & 4\\
1 & 1 & 1 & 1\\
8 & 8 & 8 & 8\\
6 & 6 & 6 & 6\\
3 & 7 & 3 & 7\\
2 & 9 & 2 & 9\\
5 & 5 & 3 & 3\\
8 & 1 & 8 & 1\\
6 & 2 & 6 & 2\\
7 & 3.6 & 7 & 3.6
\end{array}\right),\ c=\left(\begin{array}{c}
0.1\\
0.2\\
0.2\\
0.4\\
0.4\\
0.6\\
0.3\\
0.7\\
0.5\\
0.6
\end{array}\right)$. 
\begin{itemize}
\item \textbf{Sinusoidal} function defined as:
\[
f(x)=-\left(2.5\prod_{i=1}^{n}\sin\left(x_{i}-z\right)+\prod_{i=1}^{n}\sin\left(5\left(x_{i}-z\right)\right)\right),\quad0\le x_{i}\le\pi.
\]
The values of $n=4,8,16$ were used in the conducted experiments.
\item \textbf{Test2N} function:
\[
f(x)=\frac{1}{2}\sum_{i=1}^{n}x_{i}^{4}-16x_{i}^{2}+5x_{i},\quad x_{i}\in[-5,5].
\]
For the conducted experiments the values $n=4,5,6,7$ were used.
\item \textbf{Test30N} function:
\[
f(x)=\frac{1}{10}\sin^{2}\left(3\pi x_{1}\right)\sum_{i=2}^{n-1}\left(\left(x_{i}-1\right)^{2}\left(1+\sin^{2}\left(3\pi x_{i+1}\right)\right)\right)+\left(x_{n}-1\right)^{2}\left(1+\sin^{2}\left(2\pi x_{n}\right)\right)
\]
The values $n=3,4$ were used in the conducted experiments.
\end{itemize}
%

\subsection{Experimental results}

\begin{table}
\caption{Experimental settings. The numbers in cells denote the values used
in the experiments for all parameters.}

\begin{tabular*}{1\textwidth}{@{\extracolsep{\fill}}>{\centering}p{0.33\textwidth}>{\centering}p{0.33\textwidth}>{\centering}p{0.33\textwidth}}
\toprule 
 &  & \tabularnewline
\midrule 
$Nc$ & Number of chromosomes/particles & 200\tabularnewline
\midrule 
$Ng$ & Maximum number of allowed iterations & 200\tabularnewline
\midrule 
$Nm$ & Number of initial samples for K-means & 10 \texttimes{} Nc\tabularnewline
$Nk$ & Number of iterations for stopping rule & 5\tabularnewline
$ps$ & Selection rate for the genetic algorithm & 0.1\tabularnewline
$pm$ & Mutation rate for the genetic algorithm & 0.05\tabularnewline
\end{tabular*}
\end{table}

The following applies to table \ref{tab:comparison}:
\begin{itemize}
\item The column FUNCTION denotes the name of the objective problem.
\item The column GENETIC denotes the application of a genetic algorith to
the objective problem. The genetic algorithm has $N_{c}$ chromosomes
and the maximum number of allowed generations was set to $N_{g}$.
\item The column PSO stands for the application of Particle Swarm Optimizer
to every objective problem. The number of particles was set to $N_{c}$
and the maximum number of allowed iterations was set to $N_{g}$.
\item The column EEGO represents the application of the proposed method
using the values for the parameters shown in Table
\item The row SUM represents the sum of function calls for all test functions.
\end{itemize}

\subsection*{
\begin{table}[H]
\protect\caption{Experimental results using different optimization methods. Numbers
in cells represent sum function calls.\label{tab:comparison}}

\protect\centering{}%
\begin{tabular}{|c|c|c|c|}
\hline 
FUNCTION & GENETIC & PSO & EOFA\tabularnewline
\hline 
\hline 
BF1 & 4007 & 4142 & 3228\tabularnewline
\hline 
BF2 & 3794 & 3752 & 2815\tabularnewline
\hline 
BRANIN & 2376 & 2548 & 1684\tabularnewline
\hline 
CAMEL & 2869 & 2933 & 2262\tabularnewline
\hline 
EASOM & 1958 & 1982 & 1334\tabularnewline
\hline 
EXP4 & 2946 & 3404 & 2166\tabularnewline
\hline 
EXP8 & 3120 & 3585 & 2802\tabularnewline
\hline 
EXP16 & 3250 & 3735 & 3279\tabularnewline
\hline 
EXP32 & 3561 & 3902 & 3430\tabularnewline
\hline 
GKLS250 & 2280 & 2411 & 1603\tabularnewline
\hline 
GKLS350 & 2613 & 2234 & 1298\tabularnewline
\hline 
GOLDSTEIN & 3687 & 3865 & 2784\tabularnewline
\hline 
GRIEWANK2 & 4501 & 3076 & 2589\tabularnewline
\hline 
GRIEWANK10 & 6410 & 8006 & 7435\tabularnewline
\hline 
HANSEN & 3210 & 2856 & 2484\tabularnewline
\hline 
HARTMAN3 & 2752 & 3140 & 1793\tabularnewline
\hline 
HARTMAN6 & 3219 & 3710 & 2478\tabularnewline
\hline 
POTENTIAL3 & 4352 & 4865 & 4081\tabularnewline
\hline 
POTENTIAL5 & 7705 & 9183 & 8886\tabularnewline
\hline 
RASTRIGIN & 4107 & 3477 & 2304\tabularnewline
\hline 
ROSENBROCK4 & 3679 & 6372 & 4019\tabularnewline
\hline 
ROSENBROCK8 & 5270 & 8284 & 6801\tabularnewline
\hline 
ROSENBROCK16 & 8509 & 11872 & 11996\tabularnewline
\hline 
SHEKEL5 & 3325 & 4259 & 2495\tabularnewline
\hline 
SHEKEL7 & 3360 & 4241 & 2432\tabularnewline
\hline 
SHEKEL10 & 3488 & 4237 & 2516\tabularnewline
\hline 
TEST2N4 & 3331 & 3437 & 2277\tabularnewline
\hline 
TEST2N5 & 4000 & 3683 & 2734\tabularnewline
\hline 
TEST2N6 & 4312 & 3781 & 2905\tabularnewline
\hline 
TEST2N7 & 4775 & 4060 & 3559\tabularnewline
\hline 
SINU4 & 2991 & 3504 & 2005\tabularnewline
\hline 
SINU8 & 3442 & 4213 & 3158\tabularnewline
\hline 
SINU16 & 4320 & 5019 & 5891\tabularnewline
\hline 
TEST30N3 & 3211 & 4610 & 2362\tabularnewline
\hline 
TEST30N4 & 3679 & 4629 & 2978\tabularnewline
\hline 
\textbf{SUM} & 134409 & 153006 & 118863\tabularnewline
\hline 
\end{tabular}\protect
\end{table}
}

One more experiment which was performed with the ultimate goal of
measuring the importance of K-means sampling in the proposed method.
2 different sampling methods were used in this experiment:
\begin{itemize}
\item The UNIFORM column is about incorporating uniform sampling into the
genetic algorithm. Where N\_\{c\} are randomly selected chromosomes
using uniform sampling in the genetic algorithm.
\item The KMEANS column, which refers to the implementation of k-means sampling
as proposed in the genetic algorithm.
\end{itemize}
\begin{table}[H]
\caption{Experiments using different sampling techniques for the proposed method.\label{tab:sampling}}

\centering{}%
\begin{tabular}{|c|c|c|}
\hline 
FUNCTION & UNIFORM & KMEANS\tabularnewline
\hline 
\hline 
BF1 & 4513 & 3228\tabularnewline
\hline 
BF2 & 3959 & 2815\tabularnewline
\hline 
BRANIN & 2282 & 1684\tabularnewline
\hline 
CAMEL & 3156 & 2262\tabularnewline
\hline 
EASOM & 1756 & 1334\tabularnewline
\hline 
EXP4 & 3438 & 2166\tabularnewline
\hline 
EXP8 & 3432 & 2802\tabularnewline
\hline 
EXP16 & 3369 & 3279\tabularnewline
\hline 
EXP32 & 3216 & 3430\tabularnewline
\hline 
GKLS250 & 2268 & 1603\tabularnewline
\hline 
GKLS350 & 2151 & 1298\tabularnewline
\hline 
GOLDSTEIN & 3855 & 2784\tabularnewline
\hline 
GRIEWANK2 & 4310 & 2589\tabularnewline
\hline 
GRIEWANK10 & 8640 & 7435\tabularnewline
\hline 
HANSEN & 3329 & 2484\tabularnewline
\hline 
HARTMAN3 & 2849 & 1793\tabularnewline
\hline 
HARTMAN6 & 3456 & 2478\tabularnewline
\hline 
POTENTIAL3 & 4554 & 4081\tabularnewline
\hline 
POTENTIAL5 & 8356 & 8886\tabularnewline
\hline 
RASTRIGIN & 3310 & 2304\tabularnewline
\hline 
ROSENBROCK4 & 6566 & 4019\tabularnewline
\hline 
ROSENBROCK8 & 8379 & 6801\tabularnewline
\hline 
ROSENBROCK16 & 11921 & 11996\tabularnewline
\hline 
SHEKEL5 & 3946 & 2495\tabularnewline
\hline 
SHEKEL7 & 3990 & 2432\tabularnewline
\hline 
SHEKEL10 & 3836 & 2516\tabularnewline
\hline 
TEST2N4 & 3345 & 2277\tabularnewline
\hline 
TEST2N5 & 3937 & 2734\tabularnewline
\hline 
TEST2N6 & 4008 & 2905\tabularnewline
\hline 
TEST2N7 & 4545 & 3559\tabularnewline
\hline 
SINU4 & 3128 & 2005\tabularnewline
\hline 
SINU8 & 4126 & 3158\tabularnewline
\hline 
SINU16 & 6774 & 5891\tabularnewline
\hline 
TEST30N3 & 3704 & 2362\tabularnewline
\hline 
TEST30N4 & 4262 & 2978\tabularnewline
\hline 
\textbf{SUM} & 152666 & 118863\tabularnewline
\hline 
\end{tabular}
\end{table}


\subsection*{4. Conclusions}

In this particular article, some modifications were proposed to the
EEGO optimization method, which had the main purpose of improving
the efficiency as well as the speed of the global optimization algorithm.
The first modification was to periodically apply a sampling technique
based on the K-Means method\citep{kmeans-ereunhtikh-koinothta} Using
the sampling method we proposed helped to find the global minimum
with the greatest accuracy and in the least possible time. The second
proposed modification concerns the termination rule, which helps to
avoid unnecessary computational time being wasted in iterations.

Because the experimental results are very promising, efforts can be
made to develop the technique in various fields. A future extension
of the application may be the use of parallel computing techniques
to speed up the optimization process, such as, for example, the integration
of MPI\citep{MPI} or the OpenMP library\citep{OPENMP}.

\vspace{6pt}


\authorcontributions{For research articles with several authors, a short paragraph specifying
their individual contributions must be provided. The following statements
should be used ``Conceptualization, X.X. and Y.Y.; methodology, X.X.;
software, X.X.; validation, X.X., Y.Y. and Z.Z.; formal analysis,
X.X.; investigation, X.X.; resources, X.X.; data curation, X.X.; writing---original
draft preparation, X.X.; writing---review and editing, X.X.; visualization,
X.X.; supervision, X.X.; project administration, X.X.; funding acquisition,
Y.Y. All authors have read and agreed to the published version of
the manuscript.'', please turn to the \href{http://img.mdpi.org/data/contributor-role-instruction.pdf}{CRediT taxonomy}for
the term explanation. Authorship must be limited to those who have
contributed substantially to the work~reported.}

\funding{Please add: ``This research received no external funding'' or ``This
research was funded by NAME OF FUNDER grant number XXX.'' and and
``The APC was funded by XXX''. Check carefully that the details
given are accurate and use the standard spelling of funding agency
names at \url{https://search.crossref.org/funding}, any errors may
affect your future funding.}

\institutionalreview{In this section, you should add the Institutional Review Board Statement
and approval number, if relevant to your study. You might choose to
exclude this statement if the study did not require ethical approval.
Please note that the Editorial Office might ask you for further information.
Please add “The study was conducted in accordance with the Declaration
of Helsinki, and approved by the Institutional Review Board (or Ethics
Committee) of NAME OF INSTITUTE (protocol code XXX and date of approval).”
for studies involving humans. OR “The animal study protocol was approved
by the Institutional Review Board (or Ethics Committee) of NAME OF
INSTITUTE (protocol code XXX and date of approval).” for studies involving
animals. OR “Ethical review and approval were waived for this study
due to REASON (please provide a detailed justification).” OR “Not
applicable” for studies not involving humans or animals.}

\informedconsent{Any research article describing a study involving humans should contain
this statement. Please add ``Informed consent was obtained from all
subjects involved in the study.'' OR ``Patient consent was waived
due to REASON (please provide a detailed justification).'' OR ``Not
applicable'' for studies not involving humans. You might also choose
to exclude this statement if the study did not involve humans. 

Written informed consent for publication must be obtained from participating
patients who can be identified (including by the patients themselves).
Please state ``Written informed consent has been obtained from the
patient(s) to publish this paper'' if applicable.}

\dataavailability{We encourage all authors of articles published in MDPI journals to
share their research data. In this section, please provide details
regarding where data supporting reported results can be found, including
links to publicly archived datasets analyzed or generated during the
study. Where no new data were created, or where data is unavailable
due to privacy or ethical re-strictions, a statement is still required.
Suggested Data Availability Statements are available in section “MDPI
Research Data Policies” at \url{https://www.mdpi.com/ethics}. }

\acknowledgments{In this section you can acknowledge any support given which is not
covered by the author contribution or funding sections. This may include
administrative and technical support, or donations in kind (e.g.,
materials used for experiments).}

\conflictsofinterest{Declare conflicts of interest or state ``The authors declare no
conflicts of interest.'' Authors must identify and declare any personal
circumstances or interest that may be perceived as inappropriately
influencing the representation or interpretation of reported research
results. Any role of the funders in the design of the study; in the
collection, analyses or interpretation of data; in the writing of
the manuscript; or in the decision to publish the results must be
declared in this section. If there is no role, please state ``The
funders had no role in the design of the study; in the collection,
analyses, or interpretation of data; in the writing of the manuscript;
or in the decision to publish the~results''.}

\appendixtitles{no}

\appendixstart{}

\appendix

\begin{adjustwidth}{-\extralength}{0cm}{}

\reftitle{References}
\begin{thebibliography}{999}
\bibitem{search algorithm} Usman, M.J.: A survey of symbiotic organisms
search algorithms and applications. Neural Comput. Appl. 32(2), 547--566
(2020)

\bibitem{several metaheuristic algorithms }. Ezugwu, A.E., Adeleke,
O.J., Akinyelu, A.A., Viriri, S.: A conceptual comparison of several
metaheuristic algorithms on continuous optimisation problems. Neural
Comput. Appl. 32(10), 6207--6251 (2020)

\bibitem{hunting between groupers and giant moray eels in the Red Sea}Bshary,
R., Hohner, A., Ait-el-Djoudi, K., Fricke, H.: Interspecific communicative
and coordinated hunting between groupers and giant moray eels in the
Red Sea. PLoS Biol. 4(12), e431 (2006)

\bibitem{mutualism-parasitism}Wang, Y., \& DeAngelis, D. L. (2012).
A mutualism-parasitism system modeling host and parasite withmutualism
at low density. Mathematical Biosciences \& Engineering, 9(2), 431-444.

\bibitem{Competition in mutualistic systems}Addicott, J. F. (1985).
Competition in mutualistic systems. The biology of mutualism: ecology
and evolution. Croom Helm, London, UK, 217-247.

\bibitem{mutualistic}Aubier, T. G., Joron, M., \& Sherratt, T. N.
(2017). Mimicry among unequally defended prey should be mutualistic
when predators sample optimally. The American Naturalist, 189(3),
267-282.

\bibitem{Eel and grouper optimizer}Ali Mohammadzadeh, Seyedali Mirjalili.
(2024) Eel and grouper optimizer: a nature-inspired optimization algorithm.
Springer Science+Business Media, LLC, part of Springer Nature 2024

\bibitem{maths}Intriligator, M. D. (2002). Mathematical optimization
and economic theory. Society for Industrial and Applied Mathematics.

\bibitem{plhroforikh}A. Törn, M.M. Ali, S. Viitanen, Stochastic global
optimization: Problem classes and solution techniques. Journal of
Global Optimization \textbf{14}, pp. 437-447, 1999.

\bibitem{fysikhh}L. Yang, D. Robin, F. Sannibale, C. Steier, W. Wan,
Global optimization of an accelerator lattice using multiobjective
genetic algorithms, Nuclear Instruments and Methods in Physics Research
Section A: Accelerators, Spectrometers, Detectors and Associated Equipment
\textbf{609}, pp. 50-57, 2009.

\bibitem{fusikh}E. Iuliano, Global optimization of benchmark aerodynamic
cases using physics-based surrogate models, Aerospace Science and
Technology \textbf{67}, pp.273-286, 2017.

\bibitem{fusikh1}Q. Duan, S. Sorooshian, V. Gupta, Effective and
efficient global optimization for conceptual rainfall-runoff models,
Water Resources Research \textbf{28}, pp. 1015-1031 , 1992.

\bibitem{xhmeia}S. Heiles, R. L. Johnston, Global optimization of
clusters using electronic structure methods, Int. J. Quantum Chem.
\textbf{113}, pp. 2091-- 2109, 2013.

\bibitem{xhmeia1}W.H. Shin, J.K. Kim, D.S. Kim, C. Seok, GalaxyDock2:
Protein--ligand docking using beta-complex and global optimization,
J. Comput. Chem. \textbf{34}, pp. 2647-- 2656, 2013.

\bibitem{xhmeia2}A. Liwo, J. Lee, D.R. Ripoll, J. Pillardy, H. A.
Scheraga, Protein structure prediction by global optimization of a
potential energy function, Biophysics \textbf{96}, pp. 5482-5485,
1999.

\bibitem{iatrikh}Eva K. Lee, Large-Scale Optimization-Based Classification
Models in Medicine and Biology, Annals of Biomedical Engineering \textbf{35},
pp 1095-1109, 2007.

\bibitem{iatrikh1}Y. Cherruault, Global optimization in biology and
medicine, Mathematical and Computer Modelling \textbf{20}, pp. 119-132,
1994.

\bibitem{Ali}M. Montaz Ali, Charoenchai Khompatraporn, Zelda B. Zabinsky,
A Numerical Evaluation of Several Stochastic Algorithms on Selected
Continuous Global Optimization Test Problems, Journal of Global Optimization
\textbf{31}, pp 635-672, 2005. 

\bibitem{Floudas1}C.A. Floudas, P.M. Pardalos, C. Adjiman, W. Esposoto,
Z. G$\ddot{\mbox{u}}$m$\ddot{\mbox{u}}$s, S. Harding, J. Klepeis,
C. Meyer, C. Schweiger, Handbook of Test Problems in Local and Global
Optimization, Kluwer Academic Publishers, Dordrecht, 1999.

\bibitem{testfunc1}M.M. Ali and P. Kaelo, Improved particle swarm
algorithms for global optimization, Applied Mathematics and Computation
\textbf{196}, pp. 578-593, 2008.

\bibitem{testfunc2}H. Koyuncu, R. Ceylan, A PSO based approach: Scout
particle swarm algorithm for continuous global optimization problems,
Journal of Computational Design and Engineering \textbf{6}, pp. 129--142,
2019.

\bibitem{testfunc2-1}Patrick Siarry, Gérard Berthiau, François Durdin,
Jacques Haussy, ACM Transactions on Mathematical Software \textbf{23},
pp 209--228, 1997.

\bibitem{testfunc3}I.G. Tsoulos, I.E. Lagaris, GenMin: An enhanced
genetic algorithm for global optimization, Computer Physics Communications\textbf{
178, }pp. 843-851, 2008.

\bibitem{testfunc4}A. LaTorre, D. Molina, E. Osaba, J. Poyatos, J.
Del Ser, F. Herrera, A prescription of methodological guidelines for
comparing bio-inspired optimization algorithms, Swarm and Evolutionary
Computation \textbf{67}, 100973, 2021.

\bibitem{gkls}M. Gaviano, D.E. Ksasov, D. Lera, Y.D. Sergeyev, Software
for generation of classes of test functions with known local and global
minima for global optimization, ACM Trans. Math. Softw. \textbf{29},
pp. 469-480, 2003.

\bibitem{jones}J.E. Lennard-Jones, On the Determination of Molecular
Fields, Proc. R. Soc. Lond. A \textbf{ 106}, pp. 463--477, 1924.

\bibitem{APPS}Tang, J., Liu, G., Pan, Q.: A review on representative
swarm intelligence algorithms for solving optimization problems:applications
and trends. IEEE/CAA J. Autom. Sin. 8(10),1627--1643 (2021)

\bibitem{WOA}Mirjalili, S., Lewis, A.: The whale optimization algorithm.
Adv. Eng. Softw. 95, 51--67 (2016)

\bibitem{SCA}Mirjalili, S.: SCA: a sine cosine algorithm for solving
optimization problems. Knowl.-Based Syst. 96, 120--133 (2016)

\bibitem{SSA}Mirjalili, S., Gandomi, A.H., Mirjalili, S.Z., Saremi,
S., Faris, H., Mirjalili, S.M.: Salp swarm algorithm: a bio-inspired
optimizer for engineering design problems. Adv. Eng. Softw. 114, 163--191
(2017)

\bibitem{WOA1}Nasiri, J., \& Khiyabani, F. M. (2018). A whale optimization
algorithm (WOA) approach for clustering. Cogent Mathematics \& Statistics,
5(1), 1483565.

\bibitem{SSA1}Bairathi, D., \& Gopalani, D. (2019). Salp swarm algorithm
(SSA) for training feed-forward neural networks. In Soft Computing
for Problem Solving: SocProS 2017, Volume 1 (pp. 521-534). Springer
Singapore.

\bibitem{SCA1}Hao, Y., Song, L., Cui, L., \& Wang, H. (2019). A three-dimensional
geometric features-based SCA algorithm for compound faults diagnosis.
Measurement, 134, 480-491.

\bibitem{WOA2}Gharehchopogh, F. S., \& Gholizadeh, H. (2019). A comprehensive
survey: Whale Optimization Algorithm and its applications. Swarm and
Evolutionary Computation, 48, 1-24.

\bibitem{SSA2}Abualigah, L., Shehab, M., Alshinwan, M., \& Alabool,
H. (2020). Salp swarm algorithm: a comprehensive survey. Neural Computing
and Applications, 32(15), 11195-11215.

\bibitem{SCA2}Sahu, P. C., Prusty, R. C., \& Panda, S. (2022). Optimal
design of a robust FO-Multistage controller for the frequency awareness
of an islanded AC microgrid under i-SCA algorithm. International Journal
of Ambient Energy, 43(1), 2681-2693.

\bibitem{kmeans-1}Powell, M.J.D. A Tolerant Algorithm for Linearly
Constrained Optimization Calculations. Math. Program. 1989, 45, 547--566.

\bibitem{MPI}Gropp, W.; Lusk, E.; Doss, N.; Skjellum, A. A high-performance,
portable implementation of the MPI message passing interface standard.
Parallel Comput. 1996, 22, 789--828. 

\bibitem{OPENMP}Chandra, R. Parallel Programming in OpenMP; Morgan
Kaufmann: Cambridge, MA, USA, 2001.

\bibitem{MacQueen}J.B. MacQueen, Some Methods for classification
and Analysis of Multivariate Observations. Proceedings of 5th Berkeley
Symposium on Mathematical Statistics and Probability. Vol. 1. University
of California Press. pp. 281--297. MR 0214227. Zbl 0214.46201, 1967.

\bibitem{kmeans1}Y. Li, H. Wu, A clustering method based on K-means
algorithm, Physics Procedia \textbf{25}, pp. 1104-1109, 2012.

\bibitem{kmeans2}P. Arora, S. Varshney, Analysis of k-means and k-medoids
algorithm for big data, Procedia Computer Science \textbf{78}, pp.
507-512, 2016.

\bibitem{charilogis}Charilogis, V.; Tsoulos, I.G. Toward an Ideal
Particle Swarm Optimizer for Multidimensional Functions. Information
2022, 13, 217. 

\bibitem{kmeans3}Na, S., Xumin, L., \& Yong, G. (2010, April). Research
on k-means clustering algorithm: An improved k-means clustering algorithm.
In 2010 Third International Symposium on intelligent information technology
and security informatics (pp. 63-67). Ieee.

\bibitem{kmeans4}Nazeer, K. A., \& Sebastian, M. P. (2009, July).
Improving the Accuracy and Efficiency of the k-means Clustering Algorithm.
In Proceedings of the world congress on engineering (Vol. 1, pp. 1-3).
London, UK: Association of Engineers.

\bibitem{kmeans-paterrn}Ali, H. H., \& Kadhum, L. E. (2017). K-means
clustering algorithm applications in data mining and pattern recognition.
International Journal of Science and Research (IJSR), 6(8), 1577-1584.

\bibitem{kmeans-ereunhtikh-koinothta}Ahmed, M., Seraj, R., \& Islam,
S. M. S. (2020). The k-means algorithm: A comprehensive survey and
performance evaluation. Electronics, 9(8), 1295.

\bibitem{swarm1}Hassanien, A. E., \& Emary, E. (2018). Swarm intelligence:
principles, advances, and applications. CRC press.

\bibitem{swarm2}Tang, J., Liu, G., \& Pan, Q. (2021). A review on
representative swarm intelligence algorithms for solving optimization
problems: Applications and trends. IEEE/CAA Journal of Automatica
Sinica, 8(10), 1627-1643.

\bibitem{computer}Floudas, C. A., \& Pardalos, P. M. (Eds.). (2013).
State of the art in global optimization: computational methods and
applications

\bibitem{computer1}Horst, R., \& Pardalos, P. M. (Eds.). (2013).
Handbook of global optimization (Vol. 2). Springer Science \& Business
Media.

\bibitem{maths-1}Cánovas, M. J., Kruger, A., Phu, H. X., \& Théra,
M. (2020). Marco A. López, a Pioneer of Continuous Optimization in
Spain. Vietnam Journal of Mathematics, 48, 211-219.

\bibitem{maths2}Mahmoodabadi, M. J., \& Nemati, A. R. (2016). A novel
adaptive genetic algorithm for global optimization of mathematical
test functions and real-world problems. Engineering Science and Technology,
an International Journal, 19(4), 2002-2021.

\bibitem{key-maths3}Li, J., Xiao, X., Boukouvala, F., Floudas, C.
A., Zhao, B., Du, G., ... \& Liu, H. (2016). Data‐driven mathematical
modeling and global optimization framework for entire petrochemical
planning operations. AIChE Journal, 62(9), 3020-3040.

\bibitem{swarm}Brezočnik, L., Fister Jr, I., \& Podgorelec, V. (2018).
Swarm intelligence algorithms for feature selection: a review. Applied
Sciences, 8(9), 1521.

\bibitem{medicine}Houssein, E. H., Hosney, M. E., Mohamed, W. M.,
Ali, A. A., \& Younis, E. M. (2023). Fuzzy-based hunger games search
algorithm for global optimization and feature selection using medical
data. Neural Computing and Applications, 35(7), 5251-5275.

\bibitem{ewoa}Nadimi-Shahraki, M. H., Taghian, S., Mirjalili, S.,
Abualigah, L., Abd Elaziz, M., \& Oliva, D. (2021). EWOA-OPF: Effective
whale optimization algorithm to solve optimal power flow problem.
Electronics, 10(23), 2975. 

\bibitem{woa-ssa}Boursianis, A. D., Papadopoulou, M. S., Salucci,
M., Polo, A., Sarigiannidis, P., Psannis, K., ... \& Goudos, S. K.
(2021). Emerging swarm intelligence algorithms and their applications
in antenna design: the GWO, WOA, and SSA optimizers. Applied Sciences,
11(18), 8330.

\bibitem{woaa}Zhang, J., \& Wang, J. S. (2020). Improved whale optimization
algorithm based on nonlinear adaptive weight and golden sine operator.
IEEE Access, 8, 77013-77048.

\bibitem{ssaa}Wan, Y., Mao, M., Zhou, L., Zhang, Q., Xi, X., \& Zheng,
C. (2019). A novel nature-inspired maximum power point tracking (MPPT)
controller based on SSA-GWO algorithm for partially shaded photovoltaic
systems. Electronics, 8(6), 680.

\bibitem{determistic}Ion, I. G., Bontinck, Z., Loukrezis, D., Römer,
U., Lass, O., Ulbrich, S., ... \& De Gersem, H. (2018). Robust shape
optimization of electric devices based on deterministic optimization
methods and finite-element analysis with affine parametrization and
design elements. Electrical Engineering, 100(4), 2635-2647.

\bibitem{determistic1}Cuevas-Velásquez, V., Sordo-Ward, A., García-Palacios,
J. H., Bianucci, P., \& Garrote, L. (2020). Probabilistic model for
real-time flood operation of a dam based on a deterministic optimization
model. Water, 12(11), 3206.

\bibitem{determistic2}Pereyra, M., Schniter, P., Chouzenoux, E.,
Pesquet, J. C., Tourneret, J. Y., Hero, A. O., \& McLaughlin, S. (2015).
A survey of stochastic simulation and optimization methods in signal
processing. IEEE Journal of Selected Topics in Signal Processing,
10(2), 224-241.

\bibitem{stohastic}Hannah, L. A. (2015). Stochastic optimization.
International Encyclopedia of the Social \& Behavioral Sciences, 2,
473-481.

\bibitem{stohastic1}Kizielewicz, B., \& Sałabun, W. (2020). A new
approach to identifying a multi-criteria decision model based on stochastic
optimization techniques. Symmetry, 12(9), 1551.

\bibitem{stohastic2}Chen, T., Sun, Y., \& Yin, W. (2021). Solving
stochastic compositional optimization is nearly as easy as solving
stochastic optimization. IEEE Transactions on Signal Processing, 69,
4937-4948.

\bibitem{scaa}Gad, A. G. (2022). Particle swarm optimization algorithm
and its applications: a systematic review. Archives of computational
methods in engineering, 29(5), 2531-2561.

\bibitem{sca1}Zivkovic, M., Stoean, C., Chhabra, A., Budimirovic,
N., Petrovic, A., \& Bacanin, N. (2022). Novel improved salp swarm
algorithm: An application for feature selection. Sensors, 22(5), 1711.

\bibitem{swarm3}Emambocus, B. A. S., Jasser, M. B., \& Amphawan,
A. (2023). A survey on the optimization of artificial neural networks
using swarm intelligence algorithms. IEEE Access, 11, 1280-1294.

\bibitem{swarm4}Xu, M., Cao, L., Lu, D., Hu, Z., \& Yue, Y. (2023).
Application of swarm intelligence optimization algorithms in image
processing: A comprehensive review of analysis, synthesis, and optimization.
Biomimetics, 8(2), 235.

\end{thebibliography}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% for journal Sci
%\reviewreports{\\
%Reviewer 1 comments and authors' response\\
%Reviewer 2 comments and authors' response\\
%Reviewer 3 comments and authors' response
%}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\PublishersNote{}

\end{adjustwidth}{}
\end{document}
